### PR TITLE
[pdal] Update to 2.4.3

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Updates PDAL.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

**Comment by dg0yt** [comment]
Note: Not tested locally, but straightforward.

File: ports/pdal/portfile.cmake [hunk 1]
```diff
@@ -97,8 +97,6 @@ file(READ "${SOURCE_PATH}/vendor/kazhdan/PoissonRecon.h" kazhdan_license)
 string(REGEX REPLACE "^/\\*\n|\\*/.*\$" "" kazhdan_license "${kazhdan_license}")
 file(READ "${SOURCE_PATH}/vendor/lazperf/lazperf.hpp" lazperf_license)
 string(REGEX REPLACE "^/\\*\n|\\*/.*\$" "" lazperf_license "${lazperf_license}")
-file(READ "${SOURCE_PATH}/vendor/lazperf/detail/field_xyz.hpp" lazperf_detail_license)
-string(REGEX REPLACE "^/\\*\n|\\*/.*\$" "" lazperf_detail_license "${lazperf_detail_license}")
```
Comment (Cheney-W) [inline ports/pdal/portfile.cmake:101]: Why delete these?

Comment (dg0yt) [reply ports/pdal/portfile.cmake:101]: Upstream finally merged a lazperf version that moves these headers from GPL license to Apache. So this special case is gone.

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**